### PR TITLE
Option to add a custom message

### PR DESCRIPTION
**Pull request by philowest**
Related to issue #49.

Option to include a custom message in the advanced section of the config. Resolves build variables. We use this mostly to include maven variables as shown in the screen shots.    

Config:
screen shot 2015-04-07 at 9 08 08 pm

Outcome:
screen shot 2015-04-07 at 9 40 39 pm

### CODE REVIEW

**Comment by samrocketman** [comment]
Thanks for submitting this pull request.  I don't actively develop this plugin but I do maintain pull requests and releases.  I'll leave this PR open a little bit so more eyeballs can look at it.  If all turns out well then I'll make it a candidate to merge.

**Comment by samrocketman** [comment]
From my quick review and  understanding of Java the change seems okay.  I'll wait for others to review.

**Comment by Brantone** [comment]
For passing variables around : why the need for both `includeCustomMessage` and `customMessage` as opposed to just `customMessage` and do check "if not empty"??
Could still keep the checkbox boolean for property to remember to expand it in Job config.

File: src/main/java/jenkins/plugins/slack/ActiveNotifier.java [hunk 1]
```diff
@@ -275,6 +285,24 @@ public MessageBuilder appendTestSummary() {
...
+            buildVariables.putAll(build.getBuildVariables());
+            try {
+                buildVariables.putAll(build.getEnvironment(new LogTaskListener(logger, INFO)));
+            } catch (IOException e) {
+                logger.log(SEVERE, e.getMessage(), e);
+            } catch (InterruptedException e) {
+                logger.log(SEVERE, e.getMessage(), e);
+            }
+            message.append("\n");
+            message.append(replaceMacro(customMessage, new VariableResolver.ByMap<String>(buildVariables)));
```
Comment (Brantone) [inline src/main/java/jenkins/plugins/slack/ActiveNotifier.java]: What about using EnvVars `expand()` ? https://github.com/jenkinsci/jenkins/blob/76adbf922369fae214cf86445d422a424f7d309a/core/src/main/java/hudson/EnvVars.java#L382
If I understand it correctly, would also allow usage of Environment Variables?

**Comment by philowest** [comment]
Thanks for the suggestion - made the change to use the expand method of EnvVars. 

With regards to using includeCustomMessage and customMessage, this was done so that the custom message can be disabled/enabled without needing to delete it. So it's more a usability decision, but yes, it does mean that an additional variable needs to be passed around.

**Comment by samrocketman** [comment]
Since there was at least one reviewer other than myself I'm moving this into a `candidate to merge` status.  This means I'll merge your change locally, build, and test the functionality of the plugin as best I can.  I notice there are merge conflicts with the latest batch of merges.  I'll first attempt to resolve those conflicts myself if they're trivial conflicts.  If I can't do that then I'll post back for you to resolve the conflict.

**Comment by samrocketman** [comment]
I was not able/willing to resolve the conflict adequately.  Please rebase or merge in the latest master and resolve the conflicts.

**Comment by samrocketman** [comment]
I've changed my mind.  I'll open it up for code review a little longer.  Also, please update and resolve the merge conflicts.

**Comment by samrocketman** [comment]
Code style looks good.  I'm going to do some more testing locally and if all looks good then I'll merge.

**Pull request merged**